### PR TITLE
Fix Merge Regex and Max Buffer Size

### PR DESCRIPTION
**Pull request by jdfalk**
This increases the max buffer size to 10MB for larger commit logs, and adds support for bitbucket merge messages.

### CODE REVIEW

File: lib/execAsync.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const exec = require('child_process').exec;
 
 module.exports = function (command, options) {
+    options = Object.assign({}, options, { maxBuffer: (1024 * 1024 * 10) });
```
Comment (qoomon) [inline lib/execAsync.js:4]: A comment would be nice I think

File: lib/git.js [hunk 2]
```diff
@@ -10,11 +10,11 @@ const execAsync = require('./execAsync');
 async function getLastTag(
     {
         commitAnchor = 'HEAD',
-        matchPattern= '*',
+        matchPattern = '*',
```
Comment (qoomon) [inline lib/git.js:13]: please move style fixes into a separate commit

File: lib/git.js [hunk 3]
```diff
@@ -10,11 +10,11 @@ const execAsync = require('./execAsync');
...
         excludePattern,
     }
 ) {
-    return await execAsync(`git describe --tags --match=${matchPattern} ${excludePattern ? `--exclude=${excludePattern}` : ''} --no-abbrev ${commitAnchor}`, {encoding: 'UTF-8'})
+    return await execAsync(`git describe --tags --match=${matchPattern} ${excludePattern ? `--exclude=${excludePattern}` : ''} --no-abbrev ${commitAnchor}`, { encoding: 'UTF-8' })
```
Comment (qoomon) [inline lib/git.js:17]: please move style fixes into a separate commit

File: lib/git.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ const LOG_FIELD_SEPARATOR = '===LOG_FIELD_SEPARATOR===';
 
 async function getCommitLog(from, to = 'HEAD') {
     const gitLogFormat = ["%h", "%aI", "%s", "%b"].join(LOG_FIELD_SEPARATOR) + LOG_COMMIT_DELIMITER;
-    const gitLog = await execAsync(`git log --reverse --format=${gitLogFormat} ${from ? `${from}..` : ''}${to}`, {encoding: 'UTF-8'})
+    const gitLog = await execAsync(`git log --reverse --format=${gitLogFormat} ${from ? `${from}..` : ''}${to}`, { encoding: 'UTF-8' })
```
Comment (qoomon) [inline lib/git.js:33]: please move style fixes into a separate commit

File: lib/git.js [hunk 5]
```diff
@@ -58,7 +58,7 @@ async function isEqualCommit(rev1, rev2) {
         return false;
     }
 
-    const hashArray = await execAsync(`git rev-parse ${rev1}^{} ${rev2}^{}`, {encoding: 'UTF-8'})
+    const hashArray = await execAsync(`git rev-parse ${rev1}^{} ${rev2}^{}`, { encoding: 'UTF-8' })
```
Comment (qoomon) [inline lib/git.js:61]: please move style fixes into a separate commit

File: lib/commands/config.js [hunk 6]
```diff
@@ -86,7 +86,7 @@ function defaultConfig() {
 
     // regex patterns
     config.convention.msgRegex = /^(?<type>\w+)(?:\((?<scope>[^()]+)\))?(?<breaking>!)?:\s*(?<description>.+)/i;
-    config.convention.msgMergeRegex = /^Merge branch +["'](?<branch>.+)['"]/i;
+    config.convention.msgMergeRegex = /(^Merge branch)|(^Merged in) +(["'])*(?<branch>.+)(['"])*/i;
```
Comment (qoomon) [inline lib/commands/config.js:89]: I'm not sure if it's a good idea to introduce app specific config into default config. You could set this regex within the project specific `git-conventional-commits.json` file

Comment (qoomon) [reply lib/commands/config.js:89]: My primary issue is the bad readability of the regex if we introduce more than one pattern by default, furthermore the regex gets more error prone.

Comment (jdfalk) [reply lib/commands/config.js:89]: > I'm not sure if it's a good idea to introduce app specific config into default config. You could set this regex within the project specific `git-conventional-commits.json` file

My suggestion would be to put that as a commented out field in the config file and in the readme so it's more apparent that it can be changed outside of the code. 

As to app specific code, we already do that by defaulting to GitHub instead of Gitlab or something else. That said I 100% agree we shouldn't have app specific code just wasn't apparent to me that I was able to specify that in the config file.

Comment (qoomon) [reply lib/commands/config.js:89]: Damn it you were right it is not possible to override these regexpressions , however they should I think.

Comment (jdfalk) [reply lib/commands/config.js:89]: Agreed 100%. It should all be configurable from that file. I'm going to close this PR in favor of a solution that makes it configurable from JSON. 

In addition, I introduced a bug in this PR but I'm not sure how I'm causing it. Tests aren't catching it but on one repo the changes work perfectly, on the other it gives me an error:
```
TypeError: Cannot read properties of undefined (reading 'replace')
    at escapeMarkdown (/Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:226:17)
    at generateMarkdownCommit (/Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:141:31)
    at /Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:126:46
    at Array.forEach (<anonymous>)
    at /Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:125:35
    at Array.forEach (<anonymous>)
    at generateMarkdownCommitTypeSection (/Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:120:22)
    at /Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:79:38
    at Array.forEach (<anonymous>)
    at module.exports.generateMarkdown (/Users/jdfalk/repos/github/jdfalk/git-conventional-commits/lib/changelogGenerator.js:78:25)
```

File: lib/execAsync.js (hunk 1 above)
Comment (jdfalk) [reply lib/execAsync.js:4]: I think this should also be set as a command line parameter as it may make more sense for some to limit it to 1-2MB whereas others it should be 10-50MB.

**Comment by qoomon** [comment]
@jdfalk Hi i changed my mind, for now I'll add the bitbucket pattern so it is supported by default.

Just to be sure would following regex match your merge commit message `^Merged in +\S+\/\S+\/(?<branch>\S+)`? From What I found Bitbucket merge messages can look like this `Merged in team/repository/branch`

**Comment by qoomon** [comment]
@jdfalk feel free to try version `2.2.0`
